### ISSUE TITLE
Debugging fails with Meteor 1.5+

### ISSUE THREAD

**Issue by KeithGillette** [barbatus/typescript#56]
Updating to Meteor 1.5 appears to have broken TypeScript debugging for our project:
* Debugger stops at arbitrary code where no breakpoints are set 
* Debugger fails to stop where breakpoints are set (using WebStorm 2017.1.4)
* Debugger opens `app.js` instead of original TypeScript source files
* Debugger step over/into/... do not reliably work

These issues occurred with Meteor 1.5 and continue with an update today to:

```
meteor@1.5.1
barbatus:typescript@0.6.7
barbatus:typescript-compiler@0.9.7
barbatus:typescript-runtime@1.0.1
```

IDE used for debugging:

```
WebStorm 2017.1.4
Build #WS-171.4694.29, built on June 7, 2017
JRE: 1.8.0_112-release-736-b21 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.12.5
```

I don't know if it's related, but similar [issue with Meteor's Babel transpiler](https://github.com/meteor/meteor/issues/8611#issuecomment-315109080) was recently closed.

**Comment by sdarnell** [comment]
I have a github repo which is simply a `meteor create --example angular2-boilerplate` followed by addition of a trivial file, and a reference to that file that triggers exceptions on startup.
See https://github.com/sdarnell/srcmap-bug

The line numbers in the console output are wrong with meteor 1.5.1 (and from an earlier Meteor version that I'm trying to pinpoint).  The debug.ts file is simply:
```
1: console.trace('BaBoom');
2: export class Debug {
3:    public static Bang() {
4:        throw new Error("Boom!");
5:    }
6: }
7:
```
With Meteor 1.4.2.3, 1.4.3, 1.4.4, 1.4.4.1, 1.4.4.2, 1.4.4.3 the file and line numbers are:
 - BaBoom: at meteorInstall.both.debug.js (`both/debug.ts:1:9`)
 - Boom!: at Function.Debug.Bang (`both/debug.ts:4:15`)

In Meteor 1.5 and 1.5.1 they are:
 - BaBoom: at meteorInstall.both.debug.js (`both/collections/demo.collection.ts:4:87`)
 - Boom!: at Function.Debug.Bang (`both/debug.ts:6:2`)

In the first case the file name is also wrong, and the second case the line number is out by 2 (and the column is wrong).

The example above avoids any dependency on using a specific debugger - the error is visible in the server console from regular node stack traces, and I suspect that until these stack traces can be fixed it is hardly surprising that debuggers have issues.

For reference, other than the bundle changing from 1.4.4.3 to 1.5 the following packages changed:
```diff
--- a/.meteor/versions
+++ b/.meteor/versions
-babel-compiler@6.18.2
+babel-compiler@6.19.1
-blaze@2.2.0
+blaze@2.3.2
-boilerplate-generator@1.0.11
+boilerplate-generator@1.1.0
-ecmascript@0.7.3
-ecmascript-runtime@0.3.15
+dynamic-import@0.1.0
+ecmascript@0.8.0
+ecmascript-runtime@0.4.1
+ecmascript-runtime-client@0.4.1
+ecmascript-runtime-server@0.4.1
-meteor-base@1.0.4
+meteor-base@1.1.0
-minifier-js@2.0.0
-minimongo@1.0.23
+minifier-js@2.1.0
+minimongo@1.2.0
-modules@0.8.2
-modules-runtime@0.7.10
-mongo@1.1.17
+modules@0.9.0
+modules-runtime@0.8.0
+mongo@1.1.18
-promise@0.8.8
+promise@0.8.9
-spacebars@1.0.13
-spacebars-compiler@1.1.1
+spacebars@1.0.15
+spacebars-compiler@1.1.2
-standard-minifier-js@2.0.0
+standard-minifier-js@2.1.0
-ui@1.0.12
+ui@1.0.13
-webapp@1.3.15
+webapp@1.3.16
```

**Comment by sdarnell** [comment]
I've narrowed the issue that I was seeing down to a change in the bundler introduced in 1.5-beta.3 commit https://github.com/meteor/meteor/commit/e469eeec0537bd1f019a0d3cf242013830379c84, and it doesn't appear to be typescript specific so I have opened an issue with the details on the main meteor repo (see link above).

**Comment by sdarnell** [comment]
The fix for this is available in 1.5.2-beta.6 if you'd like to try it out with: `meteor update --release 1.5.2-beta.6`

**Comment by barbatus** [comment]
@sdarnell thanks for an awesome analysis :+1:
